### ISSUE TITLE
Duplicate declaration: Class[Apt] is already declared

### ISSUE THREAD

**Issue by coofercat** [MiamiOH/puppet-httpproxy#12]
When I try to use this module on an Ubuntu box, I get:

    Error: Could not retrieve catalog from remote server: Error 500 on SERVER: Server Error: Evaluation Error: Error while evaluating a Resource Statement, Duplicate declaration: Class[Apt] is already declared; cannot redeclare (file: /etc/puppetlabs/code/environments/dev/modules/httpproxy/manifests/package/apt.pp, line: 6) (file: /etc/puppetlabs/code/environments/dev/modules/httpproxy/manifests/package/apt.pp, line: 6, column: 3) on node mynode

I couldn't find a similar use of the `apt` class anywhere else in our Puppet code, but I guess there must be one somewhere. However, setting a proxy for Apt can be achieved by writing a conf.d file instead of stomping over `/etc/apt.conf`. This method means you can turn proxies on and off at will without affecting anything else that may be configured on the system.

I've created a PR for a patch to do this: https://github.com/MiamiOH/puppet-httpproxy/pull/11

**Comment by edestecd** [comment]
I suggest setting `packagemanager` to false if you will declare apt elsewhere.

**Comment by coofercat** [comment]
That does indeed 'solve' the problem, but then leaves Apt without proxy settings (and in my case makes Ubuntu servers different from their Centos counterparts).

Dropping a file in `apt.conf.d` seems like the 'correct' way to set proxy settings. It avoids ever stomping over any other changes that may be in place, or indeed clashing with some other part of Puppet which is trying to do something (but is not proxy aware).

**Comment by edestecd** [comment]
That seems like a change that should happen in the apt module.

**Comment by coofercat** [comment]
I do agree that the apt module should probably have a way to put these stacked configs in for you (a thin wrapper around the `File` resource, possibly with some validation or whatever). However, there doesn't seem to be that facility, so we're left with writing files ourselves.

**Comment by edestecd** [comment]
You can add it to the apt module. Its open source too

**Comment by coofercat** [comment]
True, although even assuming they accept the change, it won't fix this issue.
